### PR TITLE
Fix generating CoreData models in resourceSynthesizer .coreData()

### PR DESCRIPTION
**Pull request by alexfilimon**
Resolves https://github.com/tuist/tuist/issues/6200

### Short description 📝

I have realized that during project mapping, resources and CoreData models are explicitly separated in the project structure (﻿`Target.resources` and `﻿Target.coreDataModels`). However, in the `﻿SynthesizedResourceInterfaceProjectMapper` class, when generating code from templates, the search for files is done only in ﻿`Target.resources`. When generating the CoreData template, it looks for files with the ﻿`xcdatamodeld` extension, which are located in `﻿Target.coreDataModels`.

- I have modified it to search for the appropriate files for code generation in both `resources` and `coreDataModels`.
- I have also enhanced the tests in ﻿`SynthesizedResourceInterfaceProjectMapperTests`, and running these tests before my changes will result in failures.

### How to test the changes locally 🧐

1. Create a project with `.coreData()` resourceSynthesizer
2. Run the `tuist` target with the parameters (`generate`, `--no-open`, `--path /path/to/project`)
3. Make sure that the `TuistCoreData+AppCore.swift` file appears in the tuist Derived folder

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Review by fortmarek — COMMENTED** [review]
Thanks a lot for the PR! The changes look good – however, can you update the [ios_app_with_coredata](https://github.com/tuist/tuist/tree/main/fixtures/ios_app_with_coredata) fixture, so it would have failed without your changes?

**Comment by alexfilimon** [comment]
Yes, of course. @fortmarek thanks for the suggestion.
I updated the fixture ios_app_with_core_data and checked that without my fix this test was failing

File: fixtures/ios_app_with_coredata/Tuist/ResourceSynthesizers/CoreData.stencil
[the remark this answers is not in the document]
Comment (pepicrft) [reply fixtures/ios_app_with_coredata/Tuist/ResourceSynthesizers/CoreData.stencil:1]: Also, would you mind adding a `README.md` to the directory containing the fixture to document what the fixture is about?

**Review by pepicrft — APPROVED** [review]

**Review by fortmarek — APPROVED** [review]
👏

**Comment by fortmarek** [comment]
@all-contributors add @alexfilimon for code

**Pull request merged**